sklearn/utils/fixes.py

```diff
@@ -24,11 +24,18 @@
 
 from .deprecation import deprecated
 
-try:
-    from pkg_resources import parse_version  # type: ignore
-except ImportError:
-    # setuptools not installed
-    parse_version = LooseVersion  # type: ignore
+
+def parse_version(v):
+    if not set(v) <= set("0123456789."):
```

Comment:
Also maybe to be more explicit,

```suggestion
    if not set(v).issubset("0123456789."):
```

I had to check documentation to understand what <= does for sets. 

Also I would have expected `not (a <= b)` to imply `a > b`, however that's not true for sets as far as I can tell, so it is kind of confusing,
```py
>>> a = {1}
>>> b = {2}
>>> a <= b, a > b
(False, False)

Reply:
done